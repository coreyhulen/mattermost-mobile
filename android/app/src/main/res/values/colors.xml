<?xml version="1.0" encoding="utf-8"?>
<resources>
    <color name="neutral_darkest">#171717</color>
    <color name="neutral_lightest">#FFFFFF</color>
    <color name="white">#FFFFFF</color>
</resources>