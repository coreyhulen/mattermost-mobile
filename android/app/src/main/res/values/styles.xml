<resources>

    <!-- Base application theme. -->
    <style name="AppTheme" parent="Theme.AppCompat.DayNight.NoActionBar">
        <!-- Customize your theme here. -->
    </style>

    <style name="LightTheme" parent="AppTheme">
        <item name="backgroundColor">@color/neutral_lightest</item>
        <item name="splashIcon">@drawable/splash_light</item>
    </style>

    <style name="DarkTheme" parent="AppTheme">
        <item name="backgroundColor">@color/neutral_darkest</item>
        <item name="splashIcon">@drawable/splash_dark</item>
    </style>

</resources>
