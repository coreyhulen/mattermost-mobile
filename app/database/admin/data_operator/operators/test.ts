// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {Q} from '@nozbe/watermelondb';

import {MM_TABLES} from '@constants/database';
import App from '@typings/database/app';

import DatabaseManager, {DatabaseType} from '../../database_manager';
import DataOperator from '../index';
import {IsolatedEntities, OperationType} from '../types';
import {
    operateAppRecord,
    operateCustomEmojiRecord,
    operateGlobalRecord,
    operateRoleRecord,
    operateServersRecord,
    operateSystemRecord,
    operateTermsOfServiceRecord,
} from './index';

jest.mock('../../database_manager');

const {APP} = MM_TABLES.DEFAULT;

describe('*** DataOperator: Operators tests ***', () => {
    const createConnection = async (setActive = false) => {
        const dbName = 'server_schema_connection';
        const serverUrl = 'https://appv2.mattermost.com';
        const database = await DatabaseManager.createDatabaseConnection({
            shouldAddToDefaultDatabase: true,
            databaseConnection: {
                actionsEnabled: true,
                dbName,
                dbType: DatabaseType.SERVER,
                serverUrl,
            },
        });

        if (setActive) {
            await DatabaseManager.setActiveServerDatabase({
                displayName: dbName,
                serverUrl,
            });
        }

        return database;
    };

    it('=> operateAppRecord: should return an array of type App ', async () => {
        expect.assertions(3);

        const database = await DatabaseManager.getDefaultDatabase();
        expect(database).toBeTruthy();

        const preparedRecords = await operateAppRecord({
            database: database!,
            optType: OperationType.CREATE,
            value: {
                buildNumber: 'build-7',
                createdAt: 1,
                id: 'id-18',
                versionNumber: 'v-1',
            },
        });

        expect(preparedRecords).toBeTruthy();
        expect(preparedRecords!.collection.modelClass.name).toMatch('App');
    });

    it('=> operateGlobalRecord: should return an array of type Global', async () => {
        expect.assertions(3);

        const database = await DatabaseManager.getDefaultDatabase();
        expect(database).toBeTruthy();

        const preparedRecords = await operateGlobalRecord({
            database: database!,
            optType: OperationType.CREATE,
            value: {id: 'g-1', name: 'g-n1', value: 'g-v1'},
        });

        expect(preparedRecords).toBeTruthy();
        expect(preparedRecords!.collection.modelClass.name).toMatch('Global');
    });

    it('=> operateServersRecord: should return an array of type Servers', async () => {
        expect.assertions(3);

        const database = await DatabaseManager.getDefaultDatabase();
        expect(database).toBeTruthy();

        const preparedRecords = await operateServersRecord({
            database: database!,
            optType: OperationType.CREATE,
            value: {
                dbPath: 'mm-server',
                displayName: 's-displayName',
                id: 's-1',
                mentionCount: 1,
                unreadCount: 0,
                url: 'https://community.mattermost.com',
            },
        });

        expect(preparedRecords).toBeTruthy();
        expect(preparedRecords!.collection.modelClass.name).toMatch('Servers');
    });

    it('=> operateCustomEmojiRecord: should return an array of type CustomEmoji', async () => {
        expect.assertions(3);

        const database = await createConnection();
        expect(database).toBeTruthy();

        const preparedRecords = await operateCustomEmojiRecord({
            database: database!,
            optType: OperationType.CREATE,
            value: {id: 'emo-1', name: 'emoji'},
        });

        expect(preparedRecords).toBeTruthy();
        expect(preparedRecords!.collection.modelClass.name).toMatch('CustomEmoji');
    });

    it('=> operateRoleRecord: should return an array of type Role ', async () => {
        expect.assertions(3);

        const database = await createConnection();
        expect(database).toBeTruthy();

        const preparedRecords = await operateRoleRecord({
            database: database!,
            optType: OperationType.CREATE,
            value: {id: 'role-1', name: 'role-name-1', permissions: []},
        });

        expect(preparedRecords).toBeTruthy();
        expect(preparedRecords!.collection.modelClass.name).toMatch('Role');
    });

    it('=> operateSystemRecord: should return an array of type System', async () => {
        expect.assertions(3);

        const database = await createConnection();
        expect(database).toBeTruthy();

        const preparedRecords = await operateSystemRecord({
            database: database!,
            optType: OperationType.CREATE,
            value: {id: 'system-1', name: 'system-name-1', value: 'system'},
        });

        expect(preparedRecords).toBeTruthy();
        expect(preparedRecords!.collection.modelClass.name).toMatch('System');
    });

    it('=> operateTermsOfServiceRecord: should return an array of type TermsOfService ', async () => {
        expect.assertions(3);

        const database = await createConnection();
        expect(database).toBeTruthy();

        const preparedRecords = await operateTermsOfServiceRecord({
            database: database!,
            optType: OperationType.CREATE,
            value: {id: 'system-1', acceptedAt: 1},
        });

        expect(preparedRecords).toBeTruthy();
        expect(preparedRecords!.collection.modelClass.name).toMatch(
            'TermsOfService',
        );
    });

    it('=> should create a record in the App table in the default database', async () => {
        expect.assertions(2);

        // Creates a record in the App table
        await DataOperator.handleIsolatedEntity({
            optType: OperationType.CREATE,
            tableName: IsolatedEntities.APP,
            values: [
                {
                    buildNumber: 'build-1',
                    createdAt: 1,
                    id: 'id-1',
                    versionNumber: 'version-1',
                },
            ],
        });

        // Do a query and find out if the value has been registered in the App table of the default database
        const connection = await DatabaseManager.getDefaultDatabase();
        expect(connection).toBeTruthy();

        const records = (await connection!.collections.
            get(APP).
            query(Q.where('id', 'id-1')).
            fetch()) as App[];

        // We should expect to have a record returned as dictated by our query
        expect(records.length).toBe(1);
    });

    it('=> should create several records in the App table in the default database', async () => {
        expect.assertions(2);

        // Creates a record in the App table
        await DataOperator.handleIsolatedEntity({
            optType: OperationType.CREATE,
            tableName: IsolatedEntities.APP,
            values: [
                {
                    buildNumber: 'build-10',
                    createdAt: 1,
                    id: 'id-10',
                    versionNumber: 'version-10',
                },
                {
                    buildNumber: 'build-11',
                    createdAt: 1,
                    id: 'id-11',
                    versionNumber: 'version-11',
                },
                {
                    buildNumber: 'build-12',
                    createdAt: 1,
                    id: 'id-12',
                    versionNumber: 'version-12',
                },
                {
                    buildNumber: 'build-13',
                    createdAt: 1,
                    id: 'id-13',
                    versionNumber: 'version-13',
                },
            ],
        });

        // Do a query and find out if the value has been registered in the App table of the default database
        const defaultDB = await DatabaseManager.getDefaultDatabase();
        expect(defaultDB).toBeTruthy();

        const records = (await defaultDB!.collections.
            get(APP).
            query(Q.where('id', Q.oneOf(['id-10', 'id-11', 'id-12', 'id-13']))).
            fetch()) as App[];

        // We should expect to have 4 records created
        expect(records.length).toBe(4);
    });

    it('=> should update a record in the App table in the default database', async () => {
        expect.assertions(3);

        const defaultDB = await DatabaseManager.getDefaultDatabase();
        expect(defaultDB).toBeTruthy();

        // Update record having id 'id-1'
        await DataOperator.handleIsolatedEntity({
            optType: OperationType.UPDATE,
            tableName: IsolatedEntities.APP,
            values: [
                {
                    buildNumber: 'build-13-13',
                    createdAt: 1,
                    id: 'id-1',
                    versionNumber: 'version-1',
                },
            ],
        });

        const records = (await defaultDB!.collections.
            get(APP).
            query(Q.where('id', 'id-1')).
            fetch()) as App[];
        expect(records.length).toBeGreaterThan(0);

        // Verify if the buildNumber for this record has been updated
        expect(records[0].buildNumber).toMatch('build-13-13');
    });

    it('=> should update several records in the App table in the default database', async () => {
        expect.assertions(4);

        const defaultDB = await DatabaseManager.getDefaultDatabase();
        expect(defaultDB).toBeTruthy();

        // Update records having id 'id-10' and 'id-11'
        await DataOperator.handleIsolatedEntity({
            optType: OperationType.UPDATE,
            tableName: IsolatedEntities.APP,
            values: [
                {
                    buildNumber: 'build-10x',
                    createdAt: 1,
                    id: 'id-10',
                    versionNumber: 'version-10',
                },
                {
                    buildNumber: 'build-11y',
                    createdAt: 1,
                    id: 'id-11',
                    versionNumber: 'version-11',
                },
            ],
        });

        const records = (await defaultDB!.collections.
            get(APP).
            query(Q.where('id', Q.oneOf(['id-10', 'id-11']))).
            fetch()) as App[];
        expect(records.length).toBe(2);

        // Verify if the buildNumber for those two record has been updated
        expect(records[0].buildNumber).toMatch('build-10x');
        expect(records[1].buildNumber).toMatch('build-11y');
    });

    it('=> [EDGE CASE] should UPDATE instead of CREATE record for existing id', async () => {
        expect.assertions(3);

        const defaultDB = await DatabaseManager.getDefaultDatabase();
        expect(defaultDB).toBeTruthy();

        // id-10 and id-11 exist but yet the optType is CREATE.  The operator should then prepareUpdate the records instead of prepareCreate
        await DataOperator.handleIsolatedEntity({
            optType: OperationType.CREATE,
            tableName: IsolatedEntities.APP,
            values: [
                {
                    buildNumber: 'build-10x',
                    createdAt: 1,
                    id: 'id-10',
                    versionNumber: 'version-10',
                },
                {
                    buildNumber: 'build-11x',
                    createdAt: 1,
                    id: 'id-11',
                    versionNumber: 'version-11',
                },
            ],
        });

        const records = (await defaultDB!.collections.
            get(APP).
            query(Q.where('id', Q.oneOf(['id-10', 'id-11']))).
            fetch()) as App[];

        // Verify if the buildNumber for those two record has been updated
        expect(records[0].buildNumber).toMatch('build-10x');
        expect(records[1].buildNumber).toMatch('build-11x');
    });

    it('=> [EDGE CASE] should CREATE instead of UPDATE record for non-existing id', async () => {
        expect.assertions(3);

        const defaultDB = await DatabaseManager.getDefaultDatabase();
        expect(defaultDB).toBeTruthy();

        // id-15 and id-16 do not exist but yet the optType is UPDATE.  The operator should then prepareCreate the records instead of prepareUpdate
        await DataOperator.handleIsolatedEntity({
            optType: OperationType.UPDATE,
            tableName: IsolatedEntities.APP,
            values: [
                {
                    buildNumber: 'build-10x',
                    createdAt: 1,
                    id: 'id-15',
                    versionNumber: 'version-10',
                },
                {
                    buildNumber: 'build-11x',
                    createdAt: 1,
                    id: 'id-16',
                    versionNumber: 'version-11',
                },
            ],
        });

        const records = (await defaultDB!.collections.
            get(APP).
            query(Q.where('id', Q.oneOf(['id-15', 'id-16']))).
            fetch()) as App[];

        // Verify if the buildNumber for those two record has been created
        expect(records[0].buildNumber).toMatch('build-10x');
        expect(records[1].buildNumber).toMatch('build-11x');
    });

    // TODO :  new operators
});
