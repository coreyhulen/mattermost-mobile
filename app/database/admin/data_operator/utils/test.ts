// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {mockedPosts} from './mock';
import {createPostsChain, sanitizePosts} from './utils';

describe('DataOperator: Utils tests', () => {
    it('sanitizePosts: should filter between ordered and unordered posts', () => {
        const {orderedPosts, unOrderedPosts} = sanitizePosts({posts: Object.values(mockedPosts.posts), orders: mockedPosts.order});
        expect(orderedPosts.length).toBe(4);
        expect(unOrderedPosts.length).toBe(2);
    });

    it('createPostsChain: should link posts amongst each other based on order array', () => {
        const previousPostId = 'prev_xxyuoxmehne';
        const chainedOfPosts = createPostsChain({orders: mockedPosts.order, rawPosts: Object.values(mockedPosts.posts), previousPostId});

        // eslint-disable-next-line max-nested-callbacks
        const post1 = chainedOfPosts.find((post) => {
            return post.id === '8swgtrrdiff89jnsiwiip3y1eoe';
        });
        expect(post1).toBeTruthy();
        expect(post1!.prev_post_id).toBe(previousPostId);

        // eslint-disable-next-line max-nested-callbacks
        const post2 = chainedOfPosts.find((post) => {
            return post.id === '8fcnk3p1jt8mmkaprgajoxz115a';
        });
        expect(post2).toBeTruthy();
        expect(post2!.prev_post_id).toBe('8swgtrrdiff89jnsiwiip3y1eoe');

        // eslint-disable-next-line max-nested-callbacks
        const post3 = chainedOfPosts.find((post) => {
            return post.id === '3y3w3a6gkbg73bnj3xund9o5ic';
        });
        expect(post3).toBeTruthy();
        expect(post3!.prev_post_id).toBe('8fcnk3p1jt8mmkaprgajoxz115a');

        // eslint-disable-next-line max-nested-callbacks
        const post4 = chainedOfPosts.find((post) => {
            return post.id === '4btbnmticjgw7ewd3qopmpiwqw';
        });
        expect(post4).toBeTruthy();
        expect(post4!.prev_post_id).toBe('3y3w3a6gkbg73bnj3xund9o5ic');
    });
});
