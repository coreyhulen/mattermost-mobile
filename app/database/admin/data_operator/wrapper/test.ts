// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DatabaseManager from '@database/admin/database_manager';
import {createDataOperator} from '@database/admin/data_operator/wrapper';
import DatabaseConnectionException from '@database/admin/exceptions/database_connection_exception';
import {DatabaseType} from '@typings/database/enums';

jest.mock('@database/admin/database_manager');

describe('*** DataOperator Wrapper ***', () => {
    it('=> wrapper should return an instance of DataOperator ', async () => {
        expect.assertions(1);

        const serverUrl = 'https://wrapper.mattermost.com';

        // first we create the connection and save it into default database
        await DatabaseManager.createDatabaseConnection({
            configs: {
                actionsEnabled: true,
                dbName: 'community mattermost',
                dbType: DatabaseType.SERVER,
                serverUrl,
            },
            shouldAddToDefaultDatabase: true,
        });

        const dataOperator = await createDataOperator(serverUrl);

        expect(dataOperator).toBeTruthy();
    });

    it('=> wrapper should throw an error due to invalid server url', async () => {
        expect.assertions(2);

        const serverUrl = 'https://wrapper.mattermost.com';

        // first we create the connection and save it into default database
        await DatabaseManager.createDatabaseConnection({
            configs: {
                actionsEnabled: true,
                dbName: 'test_database',
                dbType: DatabaseType.SERVER,
                serverUrl,
            },
            shouldAddToDefaultDatabase: true,
        });

        await expect(createDataOperator('https://error.com')).rejects.toThrow(
            'No database has been registered with this url: https://error.com',
        );

        await expect(createDataOperator('https://error.com')).rejects.toThrow(DatabaseConnectionException);
    });
});
