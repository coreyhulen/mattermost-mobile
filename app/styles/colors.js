// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const error = '#ff6461';

export const primaryDark = '#399fff';
export const primaryLight = '#1247b1';

export const neutral = '#5c5c5c';
export const neutralDark = '#2e2e2e';
export const neutralDarker = '#1f1f1f';
export const neutralDarkest = '#171717';
export const neutralLight = '#9b9b9b';
export const neutralLighter = '#dbdbdb';
export const neutralLightest = '#ffffff';

export const darkColors = {
    authButtonBg: primaryDark,
    authButtonText: neutralDarkest,
    buttonBgDisabled: neutralDark,
    buttonTextDisabled: neutralLight,
    containerBg: neutralDarkest,
    headerText: neutralLighter,
    inputBoxBorder: neutralLight,
    inputBoxBorderError: error,
    inputBoxText: neutralLighter,
    inputBoxBgDisabled: neutralDark,
    inputBoxTextDisabled: neutralLight,
    inputBoxBorderFocused: primaryDark,
    inputBoxTextFocused: neutralLightest,
    linkText: primaryDark,
    navigationBg: neutralDarker,
    navigationText: neutralLighter,
};

export const lightColors = {
    authButtonBg: primaryLight,
    authButtonText: neutralLightest,
    buttonBgDisabled: neutralLighter,
    buttonTextDisabled: neutral,
    containerBg: neutralLightest,
    headerText: neutralDarkest,
    inputBoxBorder: neutralLighter,
    inputBoxBorderError: error,
    inputBoxText: neutralDarkest,
    inputBoxBgDisabled: neutralLighter,
    inputBoxTextDisabled: neutral,
    inputBoxBorderFocused: primaryLight,
    inputBoxTextFocused: neutralDarkest,
    linkText: primaryLight,
    navigationBg: primaryLight,
    navigationText: neutralLightest,
};