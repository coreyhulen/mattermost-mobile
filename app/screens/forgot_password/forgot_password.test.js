// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {Appearance} from 'react-native-appearance';

import {shallowWithIntl} from 'test/intl-test-helper';
import {darkColors, lightColors} from 'app/styles/colors';
import ForgotPassword from './forgot_password.js';

describe('ForgotPassword', () => {
    const actions = {
        sendPasswordResetEmail: jest.fn(),
    };

    const baseProps = {
        actions,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(<ForgotPassword {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('snapshot for error on failure of email regex', () => {
        const wrapper = shallowWithIntl(<ForgotPassword {...baseProps}/>);

        wrapper.setState({email: 'bar'});
        wrapper.instance().submitResetPassword();
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('Should call sendPasswordResetEmail', () => {
        const wrapper = shallowWithIntl(<ForgotPassword {...baseProps}/>);

        wrapper.setState({email: 'test@test.com'});
        wrapper.instance().submitResetPassword();
        expect(actions.sendPasswordResetEmail).toBeCalledWith('test@test.com');
    });

    test('match snapshot after success of sendPasswordResetEmail', async () => {
        const sendPasswordResetEmail = async () => {
            return {
                data: {},
            };
        };
        const wrapper = shallowWithIntl(
            <ForgotPassword
                {...baseProps}
                actions={{
                    ...actions,
                    sendPasswordResetEmail,
                }}
            />);

        wrapper.setState({email: 'test@test.com'});
        wrapper.instance().submitResetPassword();
        await sendPasswordResetEmail();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should show light background when user has light color scheme set', () => {
        Appearance.getColorScheme.mockImplementation(() => 'light');
        Appearance.addChangeListener.mockImplementation(() => 'light');

        const wrapper = shallowWithIntl(<ForgotPassword {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper).toHaveStyle('backgroundColor', lightColors.containerBg);
    });

    test('should show dark background when user has dark color scheme set', () => {
        Appearance.getColorScheme.mockImplementation(() => 'dark');
        Appearance.addChangeListener.mockImplementation(() => 'dark');

        const wrapper = shallowWithIntl(<ForgotPassword {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper).toHaveStyle('backgroundColor', darkColors.containerBg);
    });
});
