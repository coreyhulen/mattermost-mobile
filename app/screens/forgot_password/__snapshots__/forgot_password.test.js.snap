// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword match snapshot after success of sendPasswordResetEmail 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ffffff",
      },
    ]
  }
>
  <Connect(StatusBar) />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 15,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../dist/assets/images/logo_light.png",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#dff0d8",
            "borderColor": "#d6e9c6",
            "marginTop": 15,
            "padding": 10,
          }
        }
      >
        <FormattedText
          defaultMessage="If the account exists, a password reset email will be sent to:"
          id="password_send.link"
          style={
            Object {
              "color": "#3c763d",
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#3c763d",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          test@test.com
        </Text>
        <FormattedText
          defaultMessage="Please check your inbox."
          id="password_send.checkInbox"
          style={
            Array [
              Object {
                "color": "#3c763d",
              },
              Object {
                "paddingTop": 15,
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`ForgotPassword should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ffffff",
      },
    ]
  }
>
  <Connect(StatusBar) />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 15,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../dist/assets/images/logo_light.png",
          }
        }
      />
      <View>
        <FormattedText
          defaultMessage="To reset your password, enter the email address you used to sign up"
          id="password_send.description"
          style={
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "300",
                "lineHeight": 22,
                "marginBottom": 32,
                "textAlign": "center",
              },
              Object {
                "color": "#171717",
              },
              Object {
                "marginTop": 32,
              },
            ]
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          blurOnSubmit={false}
          disableFullscreenUI={true}
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          placeholderTextColor="#5c5c5c"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 4,
                "fontSize": 16,
                "height": 45,
                "paddingLeft": 10,
              },
              Object {
                "borderColor": "#dbdbdb",
                "color": "#171717",
              },
              Object {
                "borderWidth": 1,
              },
              Object {
                "marginBottom": 16,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Button
          containerStyle={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderRadius": 3,
                "marginBottom": 16,
                "padding": 15,
              },
              Object {
                "backgroundColor": "#1247b1",
              },
              Object {
                "backgroundColor": "#dbdbdb",
              },
              Object {},
            ]
          }
          disabled={true}
          onPress={[Function]}
        >
          <FormattedText
            defaultMessage="Reset my password"
            id="password_send.reset"
            style={
              Array [
                Object {
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "color": "#5c5c5c",
                },
              ]
            }
          />
        </Button>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`ForgotPassword should show dark background when user has dark color scheme set 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#171717",
      },
    ]
  }
>
  <Connect(StatusBar) />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 15,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../dist/assets/images/logo_dark.png",
          }
        }
      />
      <View>
        <FormattedText
          defaultMessage="To reset your password, enter the email address you used to sign up"
          id="password_send.description"
          style={
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "300",
                "lineHeight": 22,
                "marginBottom": 32,
                "textAlign": "center",
              },
              Object {
                "color": "#dbdbdb",
              },
              Object {
                "marginTop": 32,
              },
            ]
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          blurOnSubmit={false}
          disableFullscreenUI={true}
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          placeholderTextColor="#9b9b9b"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 4,
                "fontSize": 16,
                "height": 45,
                "paddingLeft": 10,
              },
              Object {
                "borderColor": "#9b9b9b",
                "color": "#dbdbdb",
              },
              Object {
                "borderWidth": 1,
              },
              Object {
                "marginBottom": 16,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Button
          containerStyle={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderRadius": 3,
                "marginBottom": 16,
                "padding": 15,
              },
              Object {
                "backgroundColor": "#399fff",
              },
              Object {
                "backgroundColor": "#2e2e2e",
              },
              Object {},
            ]
          }
          disabled={true}
          onPress={[Function]}
        >
          <FormattedText
            defaultMessage="Reset my password"
            id="password_send.reset"
            style={
              Array [
                Object {
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#171717",
                },
                Object {
                  "color": "#9b9b9b",
                },
              ]
            }
          />
        </Button>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`ForgotPassword should show light background when user has light color scheme set 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ffffff",
      },
    ]
  }
>
  <Connect(StatusBar) />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 15,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../dist/assets/images/logo_light.png",
          }
        }
      />
      <View>
        <FormattedText
          defaultMessage="To reset your password, enter the email address you used to sign up"
          id="password_send.description"
          style={
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "300",
                "lineHeight": 22,
                "marginBottom": 32,
                "textAlign": "center",
              },
              Object {
                "color": "#171717",
              },
              Object {
                "marginTop": 32,
              },
            ]
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          blurOnSubmit={false}
          disableFullscreenUI={true}
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          placeholderTextColor="#5c5c5c"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 4,
                "fontSize": 16,
                "height": 45,
                "paddingLeft": 10,
              },
              Object {
                "borderColor": "#dbdbdb",
                "color": "#171717",
              },
              Object {
                "borderWidth": 1,
              },
              Object {
                "marginBottom": 16,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Button
          containerStyle={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderRadius": 3,
                "marginBottom": 16,
                "padding": 15,
              },
              Object {
                "backgroundColor": "#1247b1",
              },
              Object {
                "backgroundColor": "#dbdbdb",
              },
              Object {},
            ]
          }
          disabled={true}
          onPress={[Function]}
        >
          <FormattedText
            defaultMessage="Reset my password"
            id="password_send.reset"
            style={
              Array [
                Object {
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "color": "#5c5c5c",
                },
              ]
            }
          />
        </Button>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`ForgotPassword snapshot for error on failure of email regex 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ffffff",
      },
    ]
  }
>
  <Connect(StatusBar) />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 15,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../dist/assets/images/logo_light.png",
          }
        }
      />
      <View>
        <FormattedText
          defaultMessage="To reset your password, enter the email address you used to sign up"
          id="password_send.description"
          style={
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "300",
                "lineHeight": 22,
                "marginBottom": 32,
                "textAlign": "center",
              },
              Object {
                "color": "#171717",
              },
              Object {
                "marginTop": 32,
              },
            ]
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          blurOnSubmit={false}
          disableFullscreenUI={true}
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          placeholderTextColor="#5c5c5c"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 4,
                "fontSize": 16,
                "height": 45,
                "paddingLeft": 10,
              },
              Object {
                "borderColor": "#dbdbdb",
                "color": "#171717",
              },
              Object {
                "borderWidth": 2,
              },
              Object {
                "borderColor": "#ff6461",
              },
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Connect(ErrorText)
          error="Please enter a valid email address."
        />
        <Button
          containerStyle={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderRadius": 3,
                "marginBottom": 16,
                "padding": 15,
              },
              Object {
                "backgroundColor": "#1247b1",
              },
              Object {
                "marginTop": 12,
              },
            ]
          }
          disabled={false}
          onPress={[Function]}
        >
          <FormattedText
            defaultMessage="Reset my password"
            id="password_send.reset"
            style={
              Array [
                Object {
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#ffffff",
                },
              ]
            }
          />
        </Button>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;
